packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -46,21 +53,38 @@ export const TokenPicker = ({
 }: TokenPickerProps) => {
   const { tokenList, tokenListWithBalances } = useTokenListContext();
   const { chainId } = useNetworkContext();
+  const signer = useEthersSigner({ chainId });
+
   const { isConnected } = useAccount();
 
   const [tokenSearchQuery, setTokenSearchQuery] = useState("");
   const [commonTokens, setCommonTokens] = useState<TokenFromTokenlist[]>(
     TOKEN_PICKER_COMMON_TOKENS[chainId]
   );
   const [debouncedQuery, setDebouncedQuery] = useState(tokenSearchQuery);
+  const [isFetchinCustomToken, setIsFetchingCustomToken] =
+    useState<boolean>(false);
+  const [filteredTokenList, setFilteredTokenList] =
+    useState<TokenWithBalance[]>(tokenList);
 
   const tokenListSearchCleanup = () => {
     setDebouncedQuery("");
     setTokenSearchQuery("");
   };
 
+  const tokenListCleanup = () => {
+    setFilteredTokenList(
+      isConnected && tokenListWithBalances ? tokenListWithBalances : tokenList
+    );
+  };
+
   const handleTokenSearchInput = (event: ChangeEvent<HTMLInputElement>) => {
-    setDebouncedQuery(event.target.value);
+    const query = event.target.value;
```

Comment:
Maybe we can trim this value to take out the empty spaces from the query, WDYT?

![image](https://github.com/SwaprHQ/stackly-ui/assets/21271189/0dfbe1b8-7eeb-4e8e-9c58-8e088c88678e)

Reply:
This is a good UX. Will fix it!